src/main/scala/com/yotpo/metorikku/metric/Metric.scala

```diff
@@ -117,28 +128,62 @@ case class Metric(configuration: Configuration, metricDir: File, metricName: Str
     }
   }
 
+   def getMaxDataframeTime(dataFrame: DataFrame, reportLagTimeColumn: Option[String],
+                                  reportLagTimeColumnUnits:Option[String]): Long ={
+    reportLagTimeColumn match {
+      case Some(timeColumn) => {
+        dataFrame.cache()
+        try {
+          reportLagTimeColumnUnits match {
+            case Some(units) => TimeUnit.valueOf(units) match {
+              case TimeUnit.SECONDS => TimeUnit.SECONDS.toMillis(dataFrame.agg({timeColumn.toString -> "max"}).collect()(0).getLong(0))
+              case TimeUnit.MILLISECONDS => TimeUnit.MILLISECONDS.toMillis(dataFrame.agg({timeColumn.toString -> "max"}).collect()(0).getLong(0))
```

Comment:
You need a default case, since there isn't a default millis in your code (config class)